internal/controller/route/handler.go

```diff
@@ -359,8 +359,11 @@ func (r *Reconciler) assembleHttpproxy(ctx context.Context, owner *routev1.Route
 		httpproxy.Spec.TCPProxy = &contourv1.TCPProxy{}
 		for _, sameRoute := range sameHostRoutes {
 			ports, err := r.getTargetPorts(ctx, &sameRoute)
+			// Ignore the route if it's not valid.
+			// This is done so that the httpproxy can be reconciled if n out m routes with same fqdn are valid (n < m)
```

Comment:
I think you can make it more clear, instead of using "n,m" use "if there are other routes that are valid with the same FQDN"
Also, please specify that by "Ignore" and "valid" in line 362, you mean there was an error while getting TargetPorts and the keyword "continue". because the word "invalid route" can confuse others as it can be invalid in many ways
suggestion: "Continue if unable to fetch TargetPorts of the route"

Reply:
Zendeh bad Navid Jan. Such a clever suggestion.

Reply:
thanks for your significant bugfix and refactors. the whole structure of code and ci is much better now.